### ISSUE TITLE
a11y: Fix keyboard navigation, aria label/role and focus display for app bar

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12774]
a11y Issue: 18, 1, 19

1. The app bar button (see below) does not show keyboard and focus presentation.
2. The pin/unpin icon is missing `aria-label` and `role` attributes.

See below for screen shot.

All of the items in the app bar should show focus appropriately and should support keyboard navigation (enter/space to activate).

Additionally, when expanded, the pin/unpin icon should support keyboard navigation and focus appropriately.

![image](https://github.com/user-attachments/assets/e425829d-a6f2-4d58-83ef-f127920b74de)

Pin/Unpin:
![image](https://github.com/user-attachments/assets/10511305-c1e5-4726-a4d0-8c490b31f624)

Issue 19: A focus indicator is not visually available as user tabs through mentioned instances using a keyboard resulting in users with learning and mobility impairments as well as those with low vision not being able to easily find out which element currently has the focus.

![image](https://github.com/user-attachments/assets/f060f6a1-80a6-467c-a8f5-2067608b5997)

**Comment by yonasberhe23** [comment]
Needs manual validation at this time as we dont have existing accessiblity tests. Follow-up ticket to create automated tests coverage https://github.com/rancher/qa-tasks/issues/1639

**Comment by IsaSih** [comment]
Navigation using tab and return work as expected in
Rancher | v2.11-1e40bc40c55a020fc67dc3c3dde47b41dc94caf1-head
Dashboard | master https://github.com/rancher/dashboard/commit/574248d1ce67bd5b8d3b6d95f716a979c22aaa9e
Rancher CLI | v2.11.0-rc.1
Helm | dev
Machine | v0.15.0-rancher125

However, the space bar clicks the burguer menu and the pin icon, but does not perform click actions in elements inside the app menu.

**Comment by aalves08** [comment]
@IsaSih given the reply from a few weeks ago do you agree we can close this? FYI @rak-phillip @nwmac
